pkg/azure/blob.go

```diff
@@ -180,14 +181,24 @@ func (c *BlobClient) BucketExists(ctx context.Context, bucketName string) (bool,
 	if err != nil {
 		return false, err
 	}
-	_, err = container.GetProperties(ctx, nil)
-	if err != nil {
-		var stgErr *azblob.StorageError
-		if errors.As(err, &stgErr) {
-			if stgErr.ErrorCode == azblob.StorageErrorCodeContainerNotFound {
+	
+	items := container.ListBlobsFlat(&azblob.ContainerListBlobsFlatOptions{
+		MaxResults: to.Ptr(int32(1)),
+	})
+	// We call next page only once since we just want to see if we get an error
+	items.NextPage(ctx)
+	if err := items.Err(); err != nil {
+		var respErr *azcore.ResponseError
+		if errors.As(err, &respErr) {
+			if respErr.ErrorCode == string(*azblob.StorageErrorCodeContainerNotFound.ToPtr()) {
 				return false, nil
 			}
-			err = stgErr
+			err = respErr
+
+			// For a container-level SASToken, we get an AuthenticationFailed when the bucket doesn't exist
+			if respErr.ErrorCode == string(azblob.StorageErrorCodeAuthenticationFailed) {
+				return false, fmt.Errorf("(Bucket might not exist) %w", err)
```

Comment:
```suggestion
				return false, fmt.Errorf("%w (Bucket name may be incorrect or non-existent)", err)
```

Reply:
The error message is a bit long. I was thinking it might not be easily seen by users if placed at the end.

Reply:
e.g
```
3ab-4b60-b393-094065902f97","error":"failed to confirm existence of 'fluxes' bucket: GET https://testfluxsaas.blob.core.windows.net/fluxes\n--------------------------------------------------------------------------------\nRESPONSE 403: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nERROR CODE: AuthenticationFailed\n--------------------------------------------------------------------------------\n﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:daf54521-101e-003f-5ceb-d8649e000000\nTime:2022-10-05T18:50:34.3475175Z</Message><AuthenticationErrorDetail>Signature did not match. String to sign used was rl\n2022-10-05T10:53:36Z\n2022-10-05T18:53:36Z\n/blob/testfluxsaas/fluxes\n\n\nhttps\n2021-06-08\nc\n\n\n\n\n\n\n</AuthenticationErrorDetail></Error>\n--------------------------------------------------------------------------------\n  (Bucket might not exist) "
```

Reply:
Thank you for pointing that out. Starting the error within parenthesis does not feel right, so potentially?

`return false, fmt.Errorf("Bucket name may be incorrect, it does not exist or caller does not have enough permissions: %w", err)`